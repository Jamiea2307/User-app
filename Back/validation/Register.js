const Joi = require("joi");

const registerValidation = (data) => {
  const schema = Joi.object({
    name: Joi.string().required(),
    email: Joi.string().min(5).required().email(),
    password: Joi.string().min(6).required(),
  });
  return schema.validate(data);
};

module.exports.registerValidation = registerValidation;
