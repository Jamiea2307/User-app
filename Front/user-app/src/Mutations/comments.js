import { gql } from "@apollo/client";

export const CREATE_COMMENT = gql`
  mutation createComment($parent: String!, $body: String!) {
    createComment(parent: $parent, body: $body)
  }
`;

export const CREATE_REPLY = gql`
  mutation createReply(
    $parentPost: String!
    $parentComment: String!
    $body: String!
  ) {
    createReply(
      parentPost: $parentPost
      parentComment: $parentComment
      body: $body
    )
  }
`;
