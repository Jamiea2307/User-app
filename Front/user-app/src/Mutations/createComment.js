import { gql } from "@apollo/client";

export const CREATE_COMMENT = gql`
  mutation createComment($parent: String!, $body: String!) {
    createComment(parent: $parent, body: $body)
  }
`;
