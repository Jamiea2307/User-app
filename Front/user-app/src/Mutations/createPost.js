import { gql } from "@apollo/client";

export const CREATE_POST = gql`
  mutation createPost($content: String!) {
    createPost(content: $content) {
      content
    }
  }
`;
