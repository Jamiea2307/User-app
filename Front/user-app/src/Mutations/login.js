import { gql } from "@apollo/client";

export const LOGIN_USERS = gql`
  mutation loginUser($email: String!, $password: String!) {
    loginUser(email: $email, password: $password) {
      email
      password
    }
  }
`;
