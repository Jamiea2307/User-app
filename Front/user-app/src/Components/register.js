import { useMutation } from "@apollo/client";

import { CREATE_USER } from "../Mutations/register";
import { useState } from "react";
import {
  UserFormBox,
  UserBox,
  SubmitBox,
  ErrorMessage,
} from "../Styles/userFormStyles";
import { registrationData } from "../Constants/registration";
import { Link } from "react-router-dom";

const Register = () => {
  const [name, setName] = useState("");
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [errorMessage, setErrorMessage] = useState("");
  const [formDisplay, setFormDisplay] = useState("block");
  const [successDisplay, setSuccessDisplay] = useState("none");
  const [createUser] = useMutation(CREATE_USER);

  const submitUser = async () => {
    try {
      await createUser({
        variables: {
          name: name,
          email: email,
          password: password,
        },
      });
    } catch (err) {
      setErrorMessage(err.message);
    }
  };

  return (
    <UserFormBox>
      <h2>{registrationData.title}</h2>
      <form
        style={{ display: formDisplay }}
        className="registrationForm"
        onSubmit={(e) => {
          e.preventDefault();
          submitUser();
        }}
      >
        <UserBox>
          <input
            type="text"
            // value={name}
            onChange={(e) => setName(e.target.value)}
          />
          <label>{registrationData.name} </label>
        </UserBox>
        <UserBox>
          <input
            type="text"
            // value={email}
            onChange={(e) => setEmail(e.target.value)}
          />
          <label>{registrationData.email} </label>
        </UserBox>
        <UserBox>
          <input
            type="password"
            // value={password}
            onChange={(e) => setPassword(e.target.value)}
          />
          <label>{registrationData.password}</label>
        </UserBox>
        <div className="errorMessage">{errorMessage}</div>
        <SubmitBox type="submit" value="Submit" />
      </form>
      <div style={{ display: successDisplay }}>
        {registrationData.successText}
      </div>
      <Link className="pageLink" to="/Login">
        {registrationData.linkText}
      </Link>
    </UserFormBox>
  );
};

export default Register;
