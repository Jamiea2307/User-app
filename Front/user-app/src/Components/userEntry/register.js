import { useState } from "react";
import { useMutation } from "@apollo/client";
import { CREATE_USER } from "../../Mutations/register";
import { UserFormBox, SubmitBox } from "../../Styles/userFormStyles";
import { registrationData } from "../../Constants/userContent";
import { Link } from "react-router-dom";
import { pathNames } from "../../Constants/pathNames";
import RegisterGratitude from "./registerGratitude";
import UserInput from "../Widgets/userInput";

const Register = () => {
  const [userDetails, setUserDetails] = useState({
    name: "",
    email: "",
    password: "",
  });
  const [errorMessage, setErrorMessage] = useState("");
  const [createUser, { data }] = useMutation(CREATE_USER);

  console.log(userDetails);

  const submitUser = async () => {
    try {
      await createUser({
        variables: {
          name: userDetails.name,
          email: userDetails.email,
          password: userDetails.password,
        },
      });
    } catch (err) {
      setErrorMessage(err.message);
    }
  };

  return !data ? (
    <UserFormBox>
      <form
        className="registrationForm"
        onSubmit={(e) => {
          e.preventDefault();
          submitUser();
        }}
      >
        <h2>{registrationData.title}</h2>
        <UserInput
          loginData={registrationData.name}
          setValue={(name) => setUserDetails({ ...userDetails, name: name })}
          type="text"
        />
        <UserInput
          loginData={registrationData.email}
          setValue={(email) => setUserDetails({ ...userDetails, email: email })}
          type="text"
        />
        <UserInput
          loginData={registrationData.password}
          setValue={(password) =>
            setUserDetails({ ...userDetails, password: password })
          }
          type="password"
        />
        <div className="errorMessage">{errorMessage}</div>
        <SubmitBox type="submit" value="Submit" />
      </form>
      <Link className="pageLink" to={pathNames.login}>
        {registrationData.linkText}
      </Link>
    </UserFormBox>
  ) : (
    <UserFormBox>
      <RegisterGratitude />
    </UserFormBox>
  );
};

export default Register;
