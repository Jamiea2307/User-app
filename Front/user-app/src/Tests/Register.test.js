import React from "react";
import Register from "../Components/userEntry/register";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import { CREATE_USER } from "../Mutations/register";
import { MockedProvider } from "@apollo/client/testing";

it("renders correctly", () => {
  const userMock = {
    request: {
      query: CREATE_USER,
      variables: {
        name: "TestUser",
        email: "TestUser@email.com",
        password: "Password123",
      },
    },
    result: {
      data: {
        createUser: true,
      },
    },
  };
  const tree = renderer
    .create(
      <MockedProvider mocks={[userMock]} addTypename={false}>
        <Router>
          <Register />
        </Router>
      </MockedProvider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
