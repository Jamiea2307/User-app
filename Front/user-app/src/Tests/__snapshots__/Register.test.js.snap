// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bdfBwQ lozkLg"
>
  <form
    className="registrationForm"
    onSubmit={[Function]}
  >
    <h2>
      Register
    </h2>
    <div
      className="sc-dlfnbm jqeTBD user-box"
    >
      <input
        onChange={[Function]}
        type="text"
      />
      <label>
        Name:
         
      </label>
    </div>
    <div
      className="sc-dlfnbm jqeTBD user-box"
    >
      <input
        onChange={[Function]}
        type="text"
      />
      <label>
        Email:
         
      </label>
    </div>
    <div
      className="sc-dlfnbm jqeTBD user-box"
    >
      <input
        onChange={[Function]}
        type="password"
      />
      <label>
        Password:
         
      </label>
    </div>
    <div
      className="errorMessage"
    >
      
    </div>
    <input
      className="sc-gsTCUz cKOPeO"
      type="submit"
      value="Submit"
    />
  </form>
  <a
    className="pageLink"
    href="/login"
    onClick={[Function]}
  >
    Login
  </a>
</div>
`;
